controllers/dnsupdate.go

```diff
@@ -377,7 +389,38 @@ func filterOutDelegateTo(delegateTo []ibclient.NameServer, fqdn string) []ibclie
 
 func (r *GslbReconciler) configureZoneDelegation(gslb *k8gbv1beta1.Gslb) (*reconcile.Result, error) {
 	clusterGeoTag := os.Getenv("CLUSTER_GEO_TAG")
+	extClusterGeoTags := os.Getenv("EXT_GSLB_CLUSTERS_GEO_TAGS")
 	infobloxGridHost := os.Getenv("INFOBLOX_GRID_HOST")
+	if r.Config.Route53Enabled {
```

Comment:
After we implement #170 (all configuration goes to one place) we can refactor this into the factory and implementations:
i.e.
```go
//DNS factory contains only logic which based on configuration returns Route53 or infoblocks or ... or error 
dnsImplementation, err := NewDnsFactory(r.config).Get()  

//somewhere in the code we just execute
//dnsImplementation contains concrete DNS logic returned by factory
err := dnsImplementation.Execute()
```
At the end we will have simpler code because we will split logic into smaller parts which we can test independently